contracts/Comptroller.sol

```diff
@@ -1079,6 +1079,35 @@ contract Comptroller is ComptrollerV7Storage, ComptrollerInterface, ComptrollerE
         require(unitroller._acceptImplementation() == 0, "change not authorized");
     }
 
+    /// @notice Delete this function after proposal 65 is executed
+    function fixBadAccruals(address[] calldata affectedUsers, uint[] calldata amounts) external {
+        require(msg.sender == admin, "Only admin can call this function"); // Only the timelock can call this function
+        require(!proposal65FixExecuted, "Already executed this one-off function"); // Require that this function is only called once
+        require(affectedUsers.length == amounts.length, "Invalid input");
+
+        // Loop variables
+        address user;
+        uint currentAccrual;
+        uint amountToSubtract;
+
+        // Iterate through all affected users
+        for (uint i = 0; i < affectedUsers.length; ++i) {
+            user = affectedUsers[i];
+            currentAccrual = compAccrued[user];
+
+            amountToSubtract = amounts[i];
+            // Prevent subtraction underflow in the case where the user has claimed and received an incorrect amount of COMP.
+            if (amountToSubtract > currentAccrual)
+                amountToSubtract = currentAccrual;
+            
+            // Subtract the bad accrual amount from what they have accrued.
+            // Users will keep whatever they have correctly accrued.
+            compAccrued[user] = sub_(currentAccrual, amountToSubtract);
+        }
+
+        proposal65FixExecuted = true; // Makes it so that this function cannot be called again
```

Comment:
As a one-off process, this works _if_ it's run on every account. If something is missed, then there would be an active concern in the protocol still. Not sure this is per se bad, but worth considering. The alternative would be to un-restrict grants on a per-user basis if this adjustment was entered. Then if something was missed, it would be blocked until this was called. Though this runs up against the concern we've seen in Prop 64 that new accounts can look like they may have tripped the bug.

Reply:
My goal is to have Compound back to normal as soon as possible. So I think this way is best.

I think the best way to ensure that we find all affected accounts is if multiple people create scripts to find all, and then submit proposal 65 when all such scripts return the same results and we're confident in the results.